src/state/SetGraphicalItem.ts

```diff
@@ -16,16 +17,30 @@ type SetCartesianGraphicalItemProps = Omit<CartesianGraphicalItemSettings, 'stac
 
 export function SetCartesianGraphicalItem(props: SetCartesianGraphicalItemProps): null {
   const dispatch = useAppDispatch();
+  const prevPropsRef = useRef<CartesianGraphicalItemSettings | null>(null);
+
   useEffect(() => {
     const settings: CartesianGraphicalItemSettings = {
       ...props,
       stackId: getNormalizedStackId(props.stackId),
     };
-    dispatch(addCartesianGraphicalItem(settings));
+
+    if (prevPropsRef.current === null) {
+      dispatch(addCartesianGraphicalItem(settings));
```

Comment:
the bug here when reproducing seems to run this first -> then cleanup `removeCartesianGraphicalItem` -> then `replaceCartesianGraphicalItem` ->

Reply:
Turns out that this happens in strict mode.

Which would be fine if it just did the usual mount + unmount + mount again, but strict mode has bonus! It will not clear refs that were set by the first render. So in browser, the ref would revert back to `null` after unmount, in strict mode it stays.

I have a repro + test for it: github.com/recharts/recharts-integ/pull/76 and I will push the same PR with a fix.